src/driver.toit

```diff
@@ -71,19 +119,46 @@ class Driver:
     waiters_.add latch
     return latch.get
 
-  run:
+
+  /**
+  Starts the message receiver task.
+
+  Function returns a latch.  When the driver is being used manually (see
+    `--auto-start` in the constructor), the latch is set when the task has fully
+    started. This is useful to ensure the task has fully started before sending
+    CFG messages that would otherwise block until a ACK/NAK was received.
+  */
+  run -> monitor.Latch:
     assert: not runner_
     adapter_.flush
-    start-periodic-nav-packets_
+    start-latch := monitor.Latch
+
+    // Start the message parser task to parse messages as they arrive.
     runner_ = task::
+      start-latch.set true
```

Comment:
In theory this latch shouldn't be necessary.
If I do a `yield`, then the task should have the time to start. No need for a latch, or so.
Also, even if the latch is necessary, we don't need to return it, but could just wait inside the `run` function. (so instead of `return start-latch` it would just be `start-latch.get`).